### ISSUE TITLE
Linking/deduplicating certificates of the same person does not use the normalized name

### ISSUE THREAD

**Issue by ThiefMaster** [Digitaler-Impfnachweis/covpass-android#117]
## Avoid duplicates
* [x] Bug is not mentioned in the [FAQ](https://digitaler-impfnachweis-app.de/faq/)
* [x] Bug is not already reported in another issue

## Technical details

- Device name: Google Pixel 5
- OS version: Android 12
- App version: 1.13.2

## Describe the bug

I just imported my dad's booster certificate in the app, and for the first two shots the name was like "Emil Mueller" (real name changed, since dummy names are fine here). For this booster shot they spelled the umlaut correctly, so it's "Emil Müller" now. However, in both certificates the normalized version of the name is "EMIL MUELLER". The birth dates are obviously the same as well.

I would expect the three vaccinations to show up for a single person in the app, but I now have two entries, one with the umlaut version and one without.

## Steps to reproduce the issue

Import multiple certificates for the same person that only differ in diacritics but have the same normalized name,

## Expected behaviour

Only a single entry for him.

## Possible Fix

Use the normalized name (+ birth date?) to group certificates together.

## Additional context

I can provide the real certificate QR codes to a maintainer if necessary, but I think it should be easily reproducible with dummy certificates which you almost certainly have for testing purposes anyway.

**Comment by thomasaugsten** [comment]
The grouping is using standardized name and also is doing trimming there needs a different in name or birthday to be different persons

**Comment by ThiefMaster** [comment]
Can you point me to the code that handles the grouping? Because I just double-checked and standardized name + birthdate are identical..

**Comment by ThiefMaster** [comment]
I just checked the 2/2 and 3/3 certs with https://github.pathcheck.org/debug.html, here's the data (again I changed name and birth date, but the normalized versions were already identical before):

```json
"Map": [
  [
    1,
    {
      "v": [
        {
          "ci": "URN:UVCI:01DE/IZ76137A/44JNDV6M9MYV02ZXY11KM5#Z",
          "co": "DE",
          "dn": 2,
          "dt": "2021-06-25",
          "is": "Robert Koch-Institut",
          "ma": "ORG-100001699",
          "mp": "EU/1/21/1529",
          "sd": 2,
          "tg": "840539006",
          "vp": "1119305005"
        }
      ],
      "dob": "1970-01-01",
      "nam": {
        "fn": "Mueller",
        "gn": "Emil",
        "fnt": "MUELLER",
        "gnt": "EMIL"
      },
      "ver": "1.0.0"
    }
  ]
]
```

```json
"Map": [
  [
    1,
    {
      "v": [
        {
          "ci": "URN:UVCI:01DE/528115600/1QAUAK5SY1F2VDIXPYASEZH2#8",
          "co": "DE",
          "dn": 3,
          "dt": "2021-11-26",
          "is": "Robert Koch-Institut",
          "ma": "ORG-100030215",
          "mp": "EU/1/20/1528",
          "sd": 3,
          "tg": "840539006",
          "vp": "1119349007"
        }
      ],
      "dob": "1970-01-01",
      "nam": {
        "fn": "Müller",
        "gn": "Emil",
        "fnt": "MUELLER",
        "gnt": "EMIL"
      },
      "ver": "1.3.0"
    }
  ]
]
```

**Comment by Ein-Tim** [comment]
@thomasaugsten

Sure that not only CWA is doing it like this?

**Comment by thomasaugsten** [comment]
Sorry wrong repo

**Comment by timokoenig** [comment]
Seems like the whole name gets compared and not only the transliterated name. This should be changed for iOS as well

https://github.com/Digitaler-Impfnachweis/covpass-android/blob/b2e3faf4b7787aecb678bcf87a299d1883568207/covpass-sdk/src/main/java/de/rki/covpass/sdk/cert/models/GroupedCertificatesList.kt#L67

**Comment by JilAuraya** [comment]
Thank you ThiefMaster for pointing this out. I will make sure this quick fix is added to one of the next releases.

**Comment by ThiefMaster** [comment]
When implementing deduplication, maybe you could also strip surrounding whitespace? I just got my booster shot in France and they accidentally put a trailing whitespace in the last name field, and instead of being stripped that ended up in the cert, so not even the Corona-Warn-App deduplicates it now...

**Comment by alexcimander** [comment]
hey @ThiefMaster,

hmm strange - whitespace trimming should be in place already. Did you try it with the CovPass-app? 👍

**Comment by ThiefMaster** [comment]
yes - see the second edit in my previous comment ;)

**Comment by hvhaugwitz** [comment]
What is the state of this issue?

**Comment by alexcimander** [comment]
Hey everyone,

with the next release 1.19 we will have an extended name matching approach that will merge certificates based on the standardized name. Besides that it is more tolerant when it comes to "second names" or "titles" within the name.

Thank you everyone!

**Comment by ThiefMaster** [comment]
Thanks! Can you confirm that the superfluous `<` will be trimmed as well?

**Comment by pucherd** [comment]
Hello @ThiefMaster, 
Yes, the < is trimmed.

The certificate matching was improved and is released in 1.19 version.

**Comment by ThiefMaster** [comment]
Indeed, works fine now :)
